opslevel/resource_opslevel_team.go

```diff
@@ -1,311 +1,322 @@
-// 		Id: opslevel.NewID(id),
-// 	}
-
-// 	if d.HasChange("name") {
-// 		input.Name = opslevel.RefOf(d.Get("name").(string))
-// 	}
-// 	if d.HasChange("responsibilities") {
-// 		input.Responsibilities = opslevel.RefOf(d.Get("responsibilities").(string))
-// 	}
-
-// 	if d.HasChange("parent") {
-// 		if parent := d.Get("parent"); parent != "" {
-// 			input.ParentTeam = opslevel.NewIdentifier(parent.(string))
-// 		} else {
-// 			input.ParentTeam = opslevel.NewIdentifier()
-// 		}
-// 	}
-
-// 	resource, err := client.UpdateTeam(input)
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	if d.HasChange("aliases") {
-// 		err = reconcileTeamAliases(d, resource, client)
-// 		if err != nil {
-// 			return err
-// 		}
-// 	}
-
-// 	if d.HasChange("member") {
-// 		membersErr := reconcileTeamMembership(d, resource, client)
-// 		if membersErr != nil {
-// 			return membersErr
-// 		}
-// 	}
-
-// 	d.Set("last_updated", timeLastUpdated())
-// 	return resourceTeamRead(d, client)
-// }
-
-// func resourceTeamDelete(d *schema.ResourceData, client *opslevel.Client) error {
-// 	id := d.Id()
-// 	err := client.DeleteTeam(id)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.SetId("")
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &TeamResource{}
+
+var _ resource.ResourceWithImportState = &TeamResource{}
+
+type TeamResource struct {
+	CommonResourceClient
+}
+
+func NewTeamResource() resource.Resource {
+	return &TeamResource{}
+}
+
+// TeamResourceModel describes the Team managed resource.
+type TeamResourceModel struct {
+	Aliases          types.List   `tfsdk:"aliases"`
+	Id               types.String `tfsdk:"id"`
+	LastUpdated      types.String `tfsdk:"last_updated"`
+	Member           []TeamMember `tfsdk:"member"`
+	Name             types.String `tfsdk:"name"`
+	Parent           types.String `tfsdk:"parent"`
+	Responsibilities types.String `tfsdk:"responsibilities"`
+}
+
+type TeamMember struct {
+	Email types.String `tfsdk:"email"`
+	Role  types.String `tfsdk:"role"`
+}
+
+func convertTeamMember(teamMember opslevel.TeamMembership) TeamMember {
+	return TeamMember{
+		Email: RequiredStringValue(teamMember.User.Email),
+		Role:  RequiredStringValue(teamMember.Role),
+	}
+}
+
+func NewTeamResourceModel(ctx context.Context, team opslevel.Team, parent string) (TeamResourceModel, diag.Diagnostics) {
```

Comment:
Shouldn't this `parent` here be part of `team opslevel.Team`?

Reply:
@davidbloss I pass in `parent` here because we need to know if the config currently uses an alias or an ID

Reply:
But shouldn't the state of `team.ParentTeam` match the provided terraform config before this call? Why can't `team.ParentTeam` be the source of truth?

Reply:
@davidbloss it should point to the same object yes, but the user is allowed to specify an ID or an alias and in order to tell what they used I pass in `parent` here.